packages/x-components/src/x-modules/search-box/components/search-input.vue

```diff
@@ -43,232 +40,262 @@
+        default: () => ['ArrowRight']
+      },
+      /**
+       * Event that retrieves the autocomplete suggestion.
+       */
+      autocompleteSuggestionsEvent: {
+        type: String as PropType<PropsWithType<XEventsTypes, Suggestion[]>>,
+        default: 'QuerySuggestionsChanged'
+      }
+    },
+    setup: function (props) {
+      useRegisterXModule(searchBoxXModule);
 
-    protected debouncedUserAcceptedAQuery!: DebouncedFunction<[string]>;
-
-    /**
-     * When event {@link XEventsTypes.UserAcceptedAQuery} or
-     * {@link SearchBoxXEvents.UserClearedQuery} are emitted the pending debounced emit
-     * {@link XEvent} `UserAcceptedAQuery` is canceled.
-     *
-     * @internal
-     */
-    @XOn(['UserAcceptedAQuery', 'UserClearedQuery'])
-    cancelDebouncedUserAcceptedAQuery(): void {
-      this.debouncedUserAcceptedAQuery?.cancel();
-    }
+      const $x = use$x();
 
-    mounted(): void {
-      if (this.autofocus) {
-        this.focusInput();
-      }
-    }
+      const { query } = useState('searchBox', ['query']);
+
+      const inputElement = ref<HTMLInputElement>();
+
+      const searchInputMessage = ref('type your query here');
+
+      const debouncedUserAcceptedAQuery = ref<DebouncedFunction<[string]>>();
+
+      /**
+       * Generates the {@link WireMetadata} object omitting the moduleName.
+       *
+       * @returns The {@link WireMetadata} object omitting the moduleName.
+       * @internal
+       */
+      const createEventMetadata = (): Omit<WireMetadata, 'moduleName'> => {
+        return {
+          target: inputElement.value,
+          feature: 'search_box'
+        };
+      };
+
+      /**
+       * Emits {@link XEventsTypes.UserAcceptedAQuery} event.
+       *
+       * @remarks It is necessary in a separated method to use it as the parameter of debounce in
+       * emitDebouncedUserAcceptedAQuery method.
+       * @internal
+       * @param query - The query that will be emitted.
+       */
+      const emitUserAcceptedAQuery = (query: string): void => {
+        $x.emit('UserAcceptedAQuery', query, createEventMetadata());
+      };
 
-    /**
-     * Emits {@link XEventsTypes.UserAcceptedAQuery} event with a debounce configured in
-     * `instantDebounceInMs` prop.
-     *
-     * @internal
-     * @param query - The query that will be emitted.
-     */
-    emitDebouncedUserAcceptedAQuery(query: string): void {
-      if (this.instant) {
-        if (!this.debouncedUserAcceptedAQuery) {
-          this.debouncedUserAcceptedAQuery = debounce(
-            this.emitUserAcceptedAQuery.bind(this),
-            this.instantDebounceInMs
-          );
+      /**
+       * Emits {@link XEventsTypes.UserAcceptedAQuery} event with a debounce configured in
+       * `instantDebounceInMs` prop.
+       *
+       * @internal
+       * @param query - The query that will be emitted.
+       */
+      const emitDebouncedUserAcceptedAQuery = (query: string): void => {
```

Comment:
Unit tests are failing because of this function. It doesn't replicate the previous behaviour. 
Here what I changed to make tests great again:

```TS
let debouncedUserAcceptedAQuery: DebouncedFunction<[string]>;
...
function emitDebouncedUserAcceptedAQuery(query: string) {
  if (props.instant) {
    if (!debouncedUserAcceptedAQuery) {
      debouncedUserAcceptedAQuery = debounce(
        emitUserAcceptedAQuery,
        props.instantDebounceInMs
      );
    }
    debouncedUserAcceptedAQuery(query);
  }
}
...
function cancelDebouncedUserAcceptedAQuery(): void {
  debouncedUserAcceptedAQuery?.cancel();
}
['UserAcceptedAQuery', 'UserClearedQuery'].forEach(event =>
  $x.on(event as XEvent).subscribe(() => debouncedUserAcceptedAQuery?.cancel())
);
```

NOTE: I have a PR in review to add a `useDebounce` composable https://github.com/empathyco/x/pull/1475, which may makes it easier to use